chcemvediet/apps/obligees/tests/test_forms.py

```diff
@@ -1,123 +1,219 @@
+        oblgs = [self._create_obligee(name=n, street=u'{} street'.format(n), city=u'{} city'.format(n), zip=u'12345', emails=u'{}@a.com'.format(n)) for n in names]
         form = self.Form(initial={u'obligee': u'ccc'})
         rendered = self._render(u'{{ form }}', form=form)
-        self.assertInHTML(u"""
-                <input class="autocomplete" data-autocomplete-url="{url}" id="id_obligee" name="obligee" type="text" value="ccc">
-                """.format(url=reverse(u'obligees:autocomplete')), rendered)
+        html = lxml.html.fromstring(rendered)
+        form_control = html.find(u'.//input'
+                                 u'[@class="form-control pln-autocomplete"]'
+                                 u'[@data-autocomplete-url="{url}"]'
+                                 u'[@name="obligee"]'
+                                 u'[@data-name="obligee"]'
+                                 u'[@type="text"]'
+                                 u'[@value="ccc"]'.format(url=reverse(u'obligees:autocomplete')))
+        self.assertIsNotNone(form_control)
+        details = html.find(u'.//div[@class="chv-obligee-widget-details "]')
+        street = details.find(u'span[@class="chv-obligee-widget-street"]')
+        self.assertEqual(street.text, u'ccc street')
+        zip = details.find(u'span[@class="chv-obligee-widget-zip"]')
+        self.assertEqual(zip.text, u'12345')
+        city = details.find(u'span[@class="chv-obligee-widget-city"]')
+        self.assertEqual(city.text, u'ccc city')
+        email = details.find(u'span[@class="chv-obligee-widget-email"]')
+        self.assertEqual(email.text, u'ccc@a.com')
 
-    @unittest.skip(u'FIXME')
     def test_submitted_with_empty_value_but_required(self):
         form = self.Form({u'obligee': u''})
         self.assertFalse(form.is_valid())
         self.assertEqual(form.errors[u'obligee'], [u'This field is required.'])
 
         rendered = self._render(u'{{ form }}', form=form)
         self.assertInHTML(u'<ul class="errorlist"><li>This field is required.</li></ul>', rendered)
-        self.assertInHTML(u"""
-                <input class="autocomplete" data-autocomplete-url="{url}" id="id_obligee" name="obligee" type="text">
-                """.format(url=reverse(u'obligees:autocomplete')), rendered)
 
-    @unittest.skip(u'FIXME')
     def test_submitted_with_empty_value_but_not_required(self):
         form = self.Form({u'obligee': u''})
         form.fields[u'obligee'].required = False
         self.assertTrue(form.is_valid())
         self.assertIsNone(form.cleaned_data[u'obligee'])
 
         rendered = self._render(u'{{ form }}', form=form)
-        self.assertInHTML(u"""
-                <input class="autocomplete" data-autocomplete-url="{url}" id="id_obligee" name="obligee" type="text">
-                """.format(url=reverse(u'obligees:autocomplete')), rendered)
+        html = lxml.html.fromstring(rendered)
+        errorlist = html.find(u'.//ul[@class="errorlist"]')
```

Comment:
@martinmacko47 Alebo mam ist az do urovne `<li>This field is required</li>` elementu.

Reply:
@viliambalaz Praveze teraz je to lepsie. Assert kontroluje, ze sa nezobrazi **ziaden** error, nie len ten jeden.